_tools/fetch_riot_data.py

```diff
@@ -0,0 +1,290 @@
+                    l.strip() for l in search_file.readlines()
+                    if re.match(r"^.*(@defgroup|@ingroup).*$", l)
+                ]
+            for idx, line in enumerate(content):
+                if "@defgroup" in line:
+                    match = re.match(regexp, line)
+                    if match is None:
+                        if multi is True:
+                            continue
+                        else:
+                            break
+                    name = match.group(2)
+                    group = match.group(1).replace("_", "__")
+                    result = { "name": f"\"{name}\"", "group": group }
+                    if parent_regexp is not None:
+                        # @ingroup should be on the next line
+                        match = re.match(parent_regexp, content[idx + 1])
+                        if match is not None:
+                            result.update({"parent": match.group(1)})
+                        else:
+                            result.update({"parent": ""})
+                    results += [result]
+                    if multi is True:
+                        continue
+                    else:
+                        break
+    return results
+
+
+def render_data_to_file(name, data):
+    """Render data to a file using a string template."""
+    if not os.path.exists(DATA_DIR):
+        os.makedirs(DATA_DIR)
+    data_file = os.path.join(DATA_DIR, f"riot_{name}.yml")
+    with open(data_file, "w") as f_dest:
+        f_dest.write("# File generated automatically, please don't edit!!\n")
+        for elem in data:
+            template = Template(TEMPLATES[name])
+            elem.update({"doc_url": RIOT_DOC_BASE_URL})
+            f_dest.write(template.substitute(**elem))
+
+
+def _subdirs(dirname, exclude=set()):
+    for _, dirnames, _ in os.walk(os.path.join(RIOTBASE, dirname)):
+        break
+    return sorted(set(dirnames) - exclude)
+
+
+def fetch_boards_data():
+    """Fetch all boards Doxygen description data (name, doxygen group)."""
+    files = [
+        "doc.txt",
+        os.path.join("include", "board.h"),
+        os.path.join("include", "periph_conf.h"),
+    ]
+    return search_data(
+        os.path.join(RIOTBASE, "boards"),
+        _subdirs("boards", exclude=set(["common"])),
+        files, r"^.*boards_([a-zA-Z0-9\-_\.]+)[ ]+(.*)$"
+    )
+
+
+def fetch_cpus_data():
+    """Fetch all cpus Doxygen description data (name, doxygen group)."""
+    files = [
+        "doc.txt",
+        os.path.join("include", "cpu.h"),
+        os.path.join("include", "cpu_conf.h"),
+        os.path.join("include", "periph_cpu.h"),
+    ]
+    cpus = search_data(
+        os.path.join(RIOTBASE, "cpu"), _subdirs("cpu"), files,
+        r"^.*cpu_([a-zA-Z0-9\-_\.]+)[ ]+(.*)$"
+    )
+    # Some cpu names/groups are not relevant and have to be removed or need
+    # cleanup
+    return [cpu for cpu in cpus if not cpu["group"].endswith("conf")]
+
```

Comment:
Just noticed that this including in the output now many 'common' CPU groups.

Reply:
Yes, I was already on it :)

Reply:
getting the CPU's was tricky. Maybe you can use here the same algorithm from #17 ?

Reply:
(sorry, I really don't get why GH doesn't refresh my tab...)

Reply:
The current implementation in this PR is quite similar (the regexp used) and allows for reusing the same logic as other queries (boards, drivers, etc), since it's just about finding Doxygen groups.